source/features/follow-file-renames.tsx

```diff
@@ -18,49 +18,53 @@ async function findRename(lastCommitOnPage: string): Promise<File[]> {
 	return files;
 }
 
-function init(): false | void {
-	const disabledPagination = select.all('.paginate-container [disabled], .paginate-container .disabled');
+async function linkify(button: HTMLButtonElement, url: GitHubURL): Promise<void | false> {
+	const isNewer = button.textContent === 'Newer';
+
+	const fromKey = isNewer ? 'previous_filename' : 'filename';
+	const toKey = isNewer ? 'filename' : 'previous_filename';
+	const sha = (isNewer ? select : select.last)([
+		'.commit .sha', // Pre "Repository refresh" layout
+		'[aria-label="Copy the full SHA"] + a'
+	])!;
+
+	const files = await findRename(sha.textContent!.trim());
+
+	for (const file of files) {
+		if (file[fromKey] === url.filePath) {
+			if (file.status === 'renamed') {
+				url.assign({
+					route: 'commits',
+					filePath: file[toKey]
+				});
+				button.replaceWith(
+					<a
+						href={String(url)}
+						aria-label={`Renamed ${isNewer ? 'to' : 'from'} ${file[toKey]}`}
+						className="btn btn-outline BtnGroup-item tooltipped tooltipped-n tooltipped-no-delay"
+					>
+						{button.textContent}
+					</a>
+				);
+			}
+
+			return;
+		}
+	}
+}
+
+function init(): void | false {
+	const disabledPagination = select.all<HTMLButtonElement>('.paginate-container [disabled], .paginate-container .disabled');
```

Comment:
Might be worth checking if .disabled is still needed. If this selector is 1y old, we should simplify it and also use `button` to avoid having to use HTMLButtonElement